### PR TITLE
Update forbidden apis with fixed executor

### PR DESCRIPTION
**Pull request by adarshsanjeev**
This adds java.util.concurrent.Executors#newFixedThreadPool(int) to the forbidden apis list. The threadpools generated by this would be non-daemon and prevent the JVM from shutting down. 
Execs#multiThreaded() marks all threads as non daemon to prevent this.

### CODE REVIEW

File: benchmarks/src/test/java/org/apache/druid/server/coordinator/BalancerStrategyBenchmark.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
   private int reservoirSize = 1;
   private double percentOfSegmentsToConsider = 100;
   private final BalancerStrategy balancerStrategy = new CostBalancerStrategy(
-      MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(1))
+      MoreExecutors.listeningDecorator(Execs.multiThreaded(1, "Test-%d"))
```
Comment (cryptoe) [inline benchmarks/src/test/java/org/apache/druid/server/coordinator/BalancerStrategyBenchmark.java:80]: stylistic nit: Can we provide betters names than test. 
May be cost_balancer_test_%d everywhere.

Comment (adarshsanjeev) [reply benchmarks/src/test/java/org/apache/druid/server/coordinator/BalancerStrategyBenchmark.java:80]: Would cost balancer be a generic enough name for all the tests? There are other tests like JettyTest

Comment (cryptoe) [reply benchmarks/src/test/java/org/apache/druid/server/coordinator/BalancerStrategyBenchmark.java:80]: Each test would be something specific. If its a general executor then we can go ahead with "Test-%s"

**Review by cryptoe — APPROVED** [review]
Failures look unrelated.

**Pull request merged**